src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,78 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
+    public static final String COMMA = ",";
+    public static final String SUPPLY = "supply";
+    public static final String BUY = "buy";
+    public static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] output = readFile(fromFileName);
+        String result = getResult(output);
+        writeToFile(result, toFileName);
+    }
+
+    private int[] readFile(String fromFileName) {
+        List<String> fileData;
+        try {
+            fileData = Files.readAllLines(new File(fromFileName).toPath());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file", e);
+        }
+
+        int supplyAmount = 0;
+        int buyAmount = 0;
+
+        for (String data : fileData) {
+            String[] values = data.split(COMMA);
+
+            if (values.length != 2) {
+                continue;
+            }
+
+            String operation = values[0];
+            String amountString = values[1];
+
+            try {
+                int value = Integer.parseInt(amountString);
+                if (operation.equals(BUY)) {
+                    buyAmount += value;
+                } else if (operation.equals(SUPPLY)) {
+                    supplyAmount += value;
+                }
+            } catch (NumberFormatException e) {
+                throw new RuntimeException("Invalid number format in file: " + amountString);
+            }
+        }
+
+        return new int[]{supplyAmount, buyAmount};
```

Comment:
The `readFile` method is doing well by reading the file and processing the data. However, the method is slightly larger than recommended. Consider refactoring to extract the logic within the for-loop into a separate method to process each line, which can help to make the code more readable and maintainable.